### PR TITLE
NEO-2084: Side navigation component improvements

### PR DESCRIPTION
**Pull request by enrique-prado**
[Link to updated components in Deploy Preview](https://deploy-preview-315--neo-react-library-storybook.netlify.app/?path=/docs/components-left-navigation--docs)

**Before tagging the team for a review, I have done the following:**
- [x] reviewed my code changes
- [x] ensured that all tests pass
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] tagged Matt if any visual changes have occurred
- [x] done an accessibility check on my work

This fixes 2 issues that are exclusively exposed in the Portal:
1. When a TopLinkItem is set to Active state it will automatically scroll into view. There is no easy way to show this in Storybook but we can verify it in the Portal (Components category).
2. We now allow default anchor behavior to take place if no custom event handler is provided. If a custom event handler is provided then we override the anchor behavior. (By doing this our Portal astro ViewTransitions work smoother and faster)

IMPORTANT: The Portal does not yet have this fix because it first need to be done in the React component. This is what this PR is for.  I later need to create a new PR that imports the latest version of the React lib that contains this fix.

### CODE REVIEW

File: src/components/LeftNav/TopLinkItem/TopLinkItem.tsx [hunk 1]
```diff
@@ -40,14 +41,24 @@ export const TopLinkItem = ({
   const ctx = useContext(LeftNavContext);
   const [isActive, setIsActive] = useState(false);
 
+  const anchorRef = createRef<HTMLAnchorElement>();
```
Comment (yangAtSpoken) [inline src/components/LeftNav/TopLinkItem/TopLinkItem.tsx:44]: I think useRef is preferred inside functional component.

File: src/components/LeftNav/LeftNav.test.tsx [hunk 2]
```diff
@@ -11,6 +11,7 @@ import {
 } from "./LeftNav.stories";
 
 describe("LeftNav", () => {
+  window.HTMLElement.prototype.scrollIntoView = vi.fn();
```
Comment (yangAtSpoken) [inline src/components/LeftNav/LeftNav.test.tsx:14]: So this works.

Comment (enrique-prado) [reply src/components/LeftNav/LeftNav.test.tsx:14]: yes, but it needs to be at the describe level.

File: src/components/LeftNav/TopLinkItem/TopLinkItem.tsx [hunk 3]
```diff
@@ -40,14 +41,24 @@ export const TopLinkItem = ({
...
     !ctx.isActiveOverride
       ? setIsActive(href === ctx.currentUrl)
       : setIsActive(false);
   }, [ctx.currentUrl, ctx.isActiveOverride, isActive, href]);
 
+  useEffect(() => {
+    if (href === ctx.currentUrl) {
+      anchorRef?.current?.scrollIntoView();
+    }
+  }, [ctx.currentUrl, anchorRef, isActive, href]);
```
Comment (yangAtSpoken) [inline src/components/LeftNav/TopLinkItem/TopLinkItem.tsx:56]: isActive not used.

**Review by yangAtSpoken — COMMENTED** [review]
some comments

File: src/components/LeftNav/TopLinkItem/TopLinkItem.tsx (hunk 1 above)
Comment (enrique-prado) [reply src/components/LeftNav/TopLinkItem/TopLinkItem.tsx:44]: noted. It works so I'm hesitant to change it :)

File: src/components/LeftNav/TopLinkItem/TopLinkItem.tsx (hunk 3 above)
Comment (enrique-prado) [reply src/components/LeftNav/TopLinkItem/TopLinkItem.tsx]: removed

File: src/components/LeftNav/LeftNav.test.tsx (hunk 2 above)
Comment (yangAtSpoken) [reply src/components/LeftNav/LeftNav.test.tsx:14]: nice

File: src/components/LeftNav/TopLinkItem/TopLinkItem.tsx (hunk 1 above)
Comment (yangAtSpoken) [reply src/components/LeftNav/TopLinkItem/TopLinkItem.tsx:44]: Please give it a try or we are creating a tech debt.

Comment (enrique-prado) [reply src/components/LeftNav/TopLinkItem/TopLinkItem.tsx:44]: ok, will give it a try.

Comment (enrique-prado) [reply src/components/LeftNav/TopLinkItem/TopLinkItem.tsx]: fixed. it works

**Review by yangAtSpoken — APPROVED** [review]

**Pull request merged**